spec/test_utils.rb

```diff
@@ -755,3 +755,26 @@ def self.check_not_equal_objects(first_object, second_object)
     expect(Hocon::Impl::PathParser.parse_path(path.render)).to eq(path)
   end
 end
+
+# For public_api_spec
+shared_examples_for "test_from_value" do
```

Comment:
Might be better to put this in with the `public_api_spec` file, since that's where it lives in the Java code.